libraries/Microsoft.Bot.Builder.Core/MiddlewareSet.cs

```diff
@@ -57,26 +52,16 @@ public MiddlewareSet Use(IMiddleware middleware)
                 //      console.print("before");
                 //      await next();
                 //      console.print("after"); 
-                // to run as expected. 
+                // to run as expected.
 
-                if (callback != null)
-                    await callback(context).ConfigureAwait(false);
 
-                return;
-            }
-
-            // Default to "No more Middleware after this"
-            async Task next()
-            {
-                // Remove the first item from the list of middleware to call,
-                // so that the next call just has the remaining items to worry about. 
-                IEnumerable<IMiddleware> remainingMiddleware = middleware.Skip(1);
-                await ReceiveActivityInternal(context, remainingMiddleware, callback).ConfigureAwait(false);
+                return callback?.Invoke(context) ?? Task.CompletedTask;
             }
 
             // Grab the current middleware, which is the 1st element in the array, and execute it            
```

Comment:
This comment is now wrong. Could you update?

Reply:
Ah, yes I missed that. Done. In fact I made the code a little more verbose (without any actual runtime cost) to make it a little more readable and applied comments on top of that.